e2e/tests/plugins/timer/timer.e2e.spec.js

```diff
@@ -0,0 +1,156 @@
+            await triggerTimerContextMenuAction(page, 'Pause');
+            await triggerTimerContextMenuAction(page, 'Restart at 0');
+            await triggerTimerContextMenuAction(page, 'Stop');
+        });
+
+        await test.step("Use Timer actions from the 3dot menu", async () => {
+            await triggerTimer3dotMenuAction(page, 'Start');
+            await triggerTimer3dotMenuAction(page, 'Pause');
+            await triggerTimer3dotMenuAction(page, 'Restart at 0');
+            await triggerTimer3dotMenuAction(page, 'Stop');
+        });
+    });
+});
+
+/**
+ * Open the timer context menu from the object tree.
+ * Expands the 'My Items' folder if it is not already expanded.
+ * @param {import('@playwright/test').Page} page
+ */
+async function openTimerContextMenu(page) {
+    const myItemsFolder = page.locator('text=Open MCT My Items >> span').nth(3);
+    const className = await myItemsFolder.getAttribute('class');
+    if (!className.includes('c-disclosure-triangle--expanded')) {
+        await myItemsFolder.click();
+    }
+
+    await page.locator(`a:has-text("Unnamed Timer")`).click({
+        button: 'right'
+    });
+}
+
+/**
+ * Trigger a timer action from the tree context menu
+ * @param {import('@playwright/test').Page} page
+ * @param {'Start' | 'Stop' | 'Pause' | 'Restart at 0'} action
+ */
+async function triggerTimerContextMenuAction(page, action) {
+    await openTimerContextMenu(page);
+    let menuOptions = page.locator('.c-menu ul');
+    await expect.soft(menuOptions).toContainText(action);
+    await page.locator(`.c-menu ul li:has-text("${action}")`).click();
+    await openTimerContextMenu(page);
+    menuOptions = page.locator('.c-menu ul');
+    await assertAvailableActions(menuOptions, action);
+}
+
+/**
+ * Trigger a timer action from the 3dot menu
+ * @param {import('@playwright/test').Page} page
+ * @param {'Start' | 'Stop' | 'Pause' | 'Restart at 0'} action
+ */
+async function triggerTimer3dotMenuAction(page, action) {
+    const threeDotMenuButton = 'button[title="More options"]';
+    await page.click(threeDotMenuButton);
+    await page.locator(`.c-menu ul li:has-text("${action}")`).click();
+
+    // FIXME: Figure out a way to not need an explicit wait here.
+    // Need to wait for 3dot menu options to update via event.
+    // eslint-disable-next-line playwright/no-wait-for-timeout
+    await page.waitForTimeout(200);
+
+    await page.click(threeDotMenuButton);
+    const menuOptions = page.locator('.c-menu ul');
+    await assertAvailableActions(menuOptions, action);
+
+    // Dismiss 3dot menu (click outside of it)
+    await page.click('.c-object-view');
+}
+
+/**
+ * Assert the available actions in the menu after performing the given action
+ * @param {import('@playwright/test').Locator} menuLocator
+ * @param {'Start' | 'Stop' | 'Pause' | 'Restart at 0'} action
+ */
+async function assertAvailableActions(menuLocator, action) {
+    switch (action) {
+    case 'Start':
```

Comment:
what does this report look like if one of these tests fails?

Reply:
It looked like a messy wall of text. With the latest changes we assert against the timer view state instead of the available menu options.